pytng/pytng.pyx

```diff
@@ -1,147 +1,1481 @@
+            tng_util_trajectory_close( & self._traj._ptr)
+            self.is_open = False
+            self._n_frames = -1
+
+    @property
+    def n_frames(self):
+        return self._n_frames
+    
+    @property
+    def n_atoms(self):
+        return self._n_particles
+
+    @property
+    def block_set(self):  # NOTE perhaps we should not expose this
+        """Dictionary where keys are available block id
+        and values are TngDataBlock instance"""
+        return self.current_step.block_set
+
+    @property
+    def block_ids(self):  # NOTE perhaps we should not expose this
+        """List of block ids available at the current frame"""
+        return list(self.current_step.block_set.keys())
+
+    @property
+    def block_names(self):
+        """List of block names available at the current frame (unordered)"""
+        block_ids = list(self.current_step.block_set.keys())
+        return [block_dictionary[id] for id in block_ids]
+
+    def get_block_by_name(self, name):
+        if self.current_step.block_set.get(
+                block_id_dictionary[name]) is None:
+            return None
+        else:
+            return self.current_step.block_set.get(
+                block_id_dictionary[name])
+
+    def get_block_values_by_name(self, name):
+        if self.current_step.block_set.get(
+                block_id_dictionary[name]) is None:
+            return None
+        else:
+            return self.current_step.block_set.get(
+                block_id_dictionary[name]).values
+
+    @property
+    def block_strides(self):
+        return [(block_dictionary[k], v)
+                for k, v in self._frame_strides.items()]
+
+    @property
+    def pos(self):
+        if self.current_step.block_set.get(TNG_TRAJ_POSITIONS) is None:
+            return None
+        else:
+            return self.current_step.block_set.get(TNG_TRAJ_POSITIONS).values
+
+    @property
+    def vel(self):
+        if self.current_step.block_set.get(TNG_TRAJ_VELOCITIES) is None:
+            return None
+        else:
+            return self.current_step.block_set.get(TNG_TRAJ_VELOCITIES).values
+
+    @property
+    def frc(self):
+        if self.current_step.block_set.get(TNG_TRAJ_FORCES) is None:
+            return None
+        else:
+            return self.current_step.block_set.get(TNG_TRAJ_FORCES).values
+
+    @property
+    def box(self):
+        if self.current_step.block_set.get(TNG_TRAJ_BOX_SHAPE) is None:
+            return None
+        else:
+            return self.current_step.block_set.get(TNG_TRAJ_BOX_SHAPE).values
+    
+    @property
+    def step(self):
+        return self.step
+
+    def read_frame(self, frame):
+        """Read a frame (integrator step) from the file,
+           modifies the state of self.block_holder to contain
+           the current blocks"""
+        
+        if frame >= self._n_frames:
+            raise ValueError("""frame specified is greater than number of steps
```

Comment:
Instead of having a space gap in the string, use two simple strings and let Python concatenate, eg
```python
raise ValueError("frame specified .... steps "
                            "in input file {}".format(self._n_frames))
```
(format or f-string interpolation happens _before_ strings are concatenated; if you need to interpolate afterwards, use parentheses(